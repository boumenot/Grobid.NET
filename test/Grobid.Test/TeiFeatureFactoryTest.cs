using System.Xml.Linq;

using ApprovalTests;
using ApprovalTests.Reporters;

using Grobid.NET;

using Xunit;

namespace Grobid.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class TeiFeatureFactoryTest
    {
        [Fact]
        public void TeiFeatureFactoryAbstract00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<div type=\"abstract\">Abstract <lb/>The advent and acceptance of massively parallel <lb/>machines has made it increasingly important to have <lb/>tools to analyze the performance of programs running on these machines. Current day performance <lb/>tools suffer from two drawbacks: they are not scalable <lb/>and they lose specific information about the user program in their attempt for generality. In this paper, <lb/>we present Projections, a scalable performance tool, <lb/>for Charm that can provide program-specific information to help the users better understand the behavior <lb/>of their programs. <lb/></div>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryAddress00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<address>Urbana, IL 61801 Urbana, IL 61801 <lb/></address>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryAffiliation00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<byline><affiliation>Department of Computer Science Department of Computer Science <lb/>University of Illinois University of Illinois <lb/></affiliation></byline>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryAuthor00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<byline><docAuthor>Amitabh B.Sinha Laxmikant V. Kale<lb/></docAuthor></byline>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryCopyright00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<note type=\"copyright\">Copyright c fl1997 Mark Lillibridge <lb/></note>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryCopyright01()
        {
            var tei = this.InsertXmlSnippetIntoTei("<note type=\"copyrights\">┬⌐ 2004 Elsevier Ireland Ltd. All rights reserved.<lb /></note>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryDate00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<date>February 1991 <lb/></date>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryDate01()
        {
            var tei = this.InsertXmlSnippetIntoTei("<date type=\"submission\">19 November 2009<lb /></date>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryDedication00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<note type=\"dedication\">Dedicated to H. Hintenberger on the occasion of his 70th birth<lb /></note>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryDegree00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<note type=\"degree\">Submitted in partial fulfillment of the requirements <lb/>for the degree of Doctor of Philosophy. <lb/>Thesis Committee: <lb/>Robert Harper, Chair <lb/>Peter Lee <lb/>John Reynolds <lb/>Luca Cardelli, DEC SRC <lb/></note>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryEmail00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<email>email: sinha@cs.uiuc.edu </email> <email>email: kale@cs.uiuc.edu <lb/></email>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryEnTitle00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<note type=\"english-title\">Simultaneous Measurement<lb /> of Hyperfine Structure, Stark-Effect<lb /> and Zeeman-Effect<lb /> of 87 RbF<lb /> with a Molecular<lb /> Beam Resonance<lb /> Apparatus<lb /></note>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryGrant00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<note type=\"grant\">This research was sponsored by the Air Force Materiel Command (AFMC) and the Defense Advanced Research Projects Agency (DARPA) under contract number, F19628-95-C-0050. </note>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryKeyword00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<keywords>Keywords: projective reconstruction, affine reconstruction, partial calibration, qualitative depth<lb/></keywords>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryIntro00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<div type=\"intro\">1 Introduction<lb/></div>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryIntro01()
        {
            var tei = this.InsertXmlSnippetIntoTei("<div type=\"introduction\">1 Introduction<lb/></div>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryOther00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<note type=\"other\">The U.S. Government is authorized to reproduce and distribute reprints for Government purposes notwithstanding <lb/>any copyright notation thereon. <lb/>The views and conclusions contained in this document are those of the author and should not be <lb/>interpreted as representing the official policies or endorsements, either expressed or implied, of the U.S. <lb/>Government. <lb/></note>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryPhone00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<note type=\"phone\">(615) 343-4111 <lb/></note>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryPubNum00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<idno>Technical Report CS91-13 <lb/></idno>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactorySubmission00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<note type=\"submission\">Submitted to the Graduate School of the <lb/></note>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryTitle00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<docTitle><titlePart>The wizard quickly jinxed the gnomes before they vaporized</titlePart></docTitle>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryTitle01()
        {
            var tei = this.InsertXmlSnippetIntoTei("<docTitle><titlePart type='main'>The wizard quickly jinxed the gnomes before they vaporized</titlePart></docTitle>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        [Fact]
        public void TeiFeatureFactoryWeb00()
        {
            var tei = this.InsertXmlSnippetIntoTei("<ptr type = \"web\" > http://ciir.cs.umass.edu/ <lb/></ptr>");

            var testSubject = new TeiFeatureFactory();
            Approvals.Verify(testSubject.Create(XDocument.Parse(tei)));
        }

        private string InsertXmlSnippetIntoTei(string s)
        {
            var tei = $@"
<?xml version='1.0' encoding='utf-8'?>
<tei>
  <text xml:lang='en'>
    <front>
     {s}
    </front>
  </text>
</tei>";
            return tei.Trim();
        }
    }
}
