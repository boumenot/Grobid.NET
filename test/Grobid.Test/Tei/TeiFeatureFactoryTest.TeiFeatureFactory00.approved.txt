Iterative I-<title>
Optimization <title>
and <title>
Simplification <title>
of <title>
@newline
Hierarchical <title>
Clusterings <title>
@newline
Technical I-<pubnum>
Report <pubnum>
CS <pubnum>
- <pubnum>
95 <pubnum>
- <pubnum>
01 <pubnum>
@newline
Doug I-<author>
Fisher <author>
@newline
Department I-<affiliation>
of <affiliation>
Computer <affiliation>
Science <affiliation>
@newline
Box I-<address>
1679 <address>
, <address>
Station <address>
B <address>
@newline
Vanderbilt I-<affiliation>
University <affiliation>
@newline
Nashville I-<address>
, <address>
TN <address>
37235 <address>
@newline
dfisher@vuse I-<email>
. <email>
vanderbilt <email>
. <email>
edu <email>
@newline
http I-<web>
: <web>
/ <web>
/ <web>
www <web>
. <web>
vuse <web>
. <web>
vanderbilt <web>
. <web>
edu <web>
/ <web>
~dfisher <web>
/ <web>
dfisher <web>
. <web>
html <web>
@newline
( I-<phone>
615 <phone>
) <phone>
343 <phone>
- <phone>
4111 <phone>
@newline
Abstract I-<abstract>
: <abstract>
Clustering <abstract>
is <abstract>
often <abstract>
used <abstract>
for <abstract>
discovering <abstract>
structure <abstract>
in <abstract>
data <abstract>
. <abstract>
Clustering <abstract>
systems <abstract>
@newline
differ <abstract>
in <abstract>
the <abstract>
objective <abstract>
function <abstract>
used <abstract>
to <abstract>
evaluate <abstract>
clustering <abstract>
quality <abstract>
and <abstract>
the <abstract>
control <abstract>
strategy <abstract>
@newline
used <abstract>
to <abstract>
search <abstract>
the <abstract>
space <abstract>
of <abstract>
clusterings <abstract>
. <abstract>
Ideally <abstract>
, <abstract>
the <abstract>
search <abstract>
strategy <abstract>
should <abstract>
consistently <abstract>
@newline
construct <abstract>
clusterings <abstract>
of <abstract>
high <abstract>
quality <abstract>
, <abstract>
but <abstract>
be <abstract>
computationally <abstract>
inexpensive <abstract>
as <abstract>
well <abstract>
. <abstract>
In <abstract>
general <abstract>
, <abstract>
@newline
we <abstract>
cannot <abstract>
have <abstract>
it <abstract>
both <abstract>
ways <abstract>
, <abstract>
but <abstract>
we <abstract>
can <abstract>
partition <abstract>
the <abstract>
search <abstract>
so <abstract>
that <abstract>
a <abstract>
system <abstract>
inexpensively <abstract>
@newline
constructs <abstract>
a <abstract>
` <abstract>
tentative <abstract>
' <abstract>
clustering <abstract>
for <abstract>
initial <abstract>
examination <abstract>
, <abstract>
followed <abstract>
by <abstract>
iterative <abstract>
optimization <abstract>
, <abstract>
@newline
which <abstract>
continues <abstract>
to <abstract>
search <abstract>
in <abstract>
background <abstract>
for <abstract>
improved <abstract>
clusterings <abstract>
. <abstract>
Given <abstract>
this <abstract>
motivation <abstract>
, <abstract>
we <abstract>
@newline
evaluate <abstract>
an <abstract>
inexpensive <abstract>
strategy <abstract>
for <abstract>
creating <abstract>
initial <abstract>
clusterings <abstract>
, <abstract>
coupled <abstract>
with <abstract>
several <abstract>
control <abstract>
@newline
strategies <abstract>
for <abstract>
iterative <abstract>
optimization <abstract>
, <abstract>
each <abstract>
of <abstract>
which <abstract>
repeatedly <abstract>
modifies <abstract>
an <abstract>
initial <abstract>
clustering <abstract>
@newline
in <abstract>
search <abstract>
of <abstract>
a <abstract>
better <abstract>
one <abstract>
. <abstract>
One <abstract>
of <abstract>
these <abstract>
methods <abstract>
appears <abstract>
novel <abstract>
as <abstract>
an <abstract>
iterative <abstract>
optimization <abstract>
@newline
strategy <abstract>
in <abstract>
clustering <abstract>
contexts <abstract>
. <abstract>
Once <abstract>
a <abstract>
clustering <abstract>
has <abstract>
been <abstract>
constructed <abstract>
it <abstract>
is <abstract>
judged <abstract>
by <abstract>
@newline
analysts <abstract>
often <abstract>
according <abstract>
to <abstract>
task <abstract>
- <abstract>
specific <abstract>
criteria <abstract>
. <abstract>
Several <abstract>
authors <abstract>
have <abstract>
abstracted <abstract>
these <abstract>
@newline
criteria <abstract>
and <abstract>
posited <abstract>
a <abstract>
generic <abstract>
performance <abstract>
task <abstract>
akin <abstract>
to <abstract>
pattern <abstract>
completion <abstract>
, <abstract>
where <abstract>
the <abstract>
error <abstract>
@newline
rate <abstract>
over <abstract>
completed <abstract>
patterns <abstract>
is <abstract>
used <abstract>
to <abstract>
' <abstract>
externally <abstract>
' <abstract>
judge <abstract>
clustering <abstract>
utility <abstract>
. <abstract>
Given <abstract>
this <abstract>
@newline
performance <abstract>
task <abstract>
we <abstract>
adapt <abstract>
resampling <abstract>
- <abstract>
based <abstract>
pruning <abstract>
strategies <abstract>
used <abstract>
by <abstract>
supervised <abstract>
learning <abstract>
@newline
systems <abstract>
to <abstract>
the <abstract>
task <abstract>
of <abstract>
simplifying <abstract>
hierarchical <abstract>
clusterings <abstract>
, <abstract>
thus <abstract>
promising <abstract>
to <abstract>
ease <abstract>
post <abstract>
- <abstract>
clustering <abstract>
analysis <abstract>
. <abstract>
Finally <abstract>
, <abstract>
we <abstract>
propose <abstract>
a <abstract>
number <abstract>
of <abstract>
objective <abstract>
functions <abstract>
, <abstract>
based <abstract>
on <abstract>
attribute <abstract>
- <abstract>
selection <abstract>
measures <abstract>
for <abstract>
decision <abstract>
- <abstract>
tree <abstract>
induction <abstract>
, <abstract>
that <abstract>
might <abstract>
perform <abstract>
well <abstract>
on <abstract>
the <abstract>
error <abstract>
rate <abstract>
and <abstract>
@newline
simplicity <abstract>
dimensions <abstract>
. <abstract>
@newline
Keywords I-<keyword>
: <keyword>
clustering <keyword>
, <keyword>
iterative <keyword>
optimization <keyword>
, <keyword>
cluster <keyword>
validation <keyword>
, <keyword>
resampling <keyword>
, <keyword>
pruning <keyword>
, <keyword>
objective <keyword>
functions <keyword>
. <keyword>
@newline